app/src/App.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from "react";
+import "./App.css";
+
+const createRow = number => {
+  const row = [];
+  let i = 0;
+  while (i < number) {
+    row.push(false);
+    i++;
+  }
+  return row;
+};
+
+function App() {
+  const [grid, setGrid] = useState(createRow(5000));
```

Comment:
`5000` seems to be a "magic number" - as per Monarch's comment it's best to define these as constants with descriptive names.